src/queries/rum-content-requests.sql

```diff
@@ -1,40 +1,306 @@
---- description: List number of content requests per day and hostname
---- startdate: 2022-01-01
---- enddate: 2022-01-02
---- timezone: UTC
+--- description: Get ContentRequests.
+--- Authorization: none
+--- Access-Control-Allow-Origin: *
+--- startdate: 2024-01-01
+--- enddate: 2024-03-31
+--- url: -
+--- granularity: 30
+--- after: -
 --- limit: 1000
---- offset: 0
 --- domainkey: secret
-SELECT
-  day,
-  contentrequests,
-  hostname
-FROM (
+WITH all_raw_events AS (
   SELECT
+    id,
+    checkpoint,
+    source,
+    target,
+    weight,
+    url,
+    host,
     hostname,
-    ROW_NUMBER() OVER (ORDER BY hostname) AS resrow,
-    FORMAT_TIMESTAMP('%F', TIMESTAMP_TRUNC(time, DAY, @timezone)) AS day,
-    SUM(pageviews) AS contentrequests
+    user_agent,
+    time,
+    TIMESTAMP_TRUNC(time, DAY, 'UTC') AS trunc_time # events will be evaluated on day level
   FROM
-    helix_rum.PAGEVIEWS_V5(
-      '-', # url
-      CAST(@offset AS INT64), # offset
-      -1, # days to fetch
+    helix_rum.EVENTS_V5(
+      @url, # list of urls
+      CAST(-1 AS INT64), # offset
+      CAST(-1 AS INT64), # days to fetch
       @startdate, # start date
       @enddate, # end date
-      @timezone, # timezone
-      'nobot', # deviceclass
-      @domainkey # domainkey
+      'UTC', # timezone
+      'all', # deviceclass
+      @domainkey # domain key to prevent data sharing
     )
-  WHERE (
-    hostname NOT LIKE '%.hlx.live'
-    AND hostname NOT LIKE '%.hlx3.live'
+  WHERE
+    hostname != ''
+    AND NOT REGEXP_CONTAINS(hostname, r'^\d+\.\d+\.\d+\.\d+$') -- IP addresses
+    AND hostname NOT LIKE 'localhost%'
     AND hostname NOT LIKE '%.hlx.page'
     AND hostname NOT LIKE '%.hlx3.page'
-    AND hostname NOT LIKE 'localhost'
-    AND hostname NOT LIKE '127.0.0.1'
-  )
-  GROUP BY day, hostname
-  ORDER BY hostname, day
+    AND hostname NOT LIKE '%.hlx.live'
+    AND hostname NOT LIKE '%.helix3.dev'
+    AND hostname NOT LIKE '%.sharepoint.com'
+    AND hostname NOT LIKE '%.aem.page'
+    AND hostname NOT LIKE '%.aem.live'
+),
+
+get_top_host AS (
```

Comment:
Nitpick: this sounds like a function, but isn't. Maybe `top_hosts` would be sufficient.